internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -893,6 +925,106 @@ func (r *BMACReconciler) ensureSpokeMachine(ctx context.Context, log logrus.Fiel
 	return machineSpoke, nil
 }
 
+// Get MCS cert of the spoke cluster, encode it and add it to BMH
+func (r *BMACReconciler) ensureMCSCert(ctx context.Context, log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
+	// Only worker role is supported for day2 operation
+	if agent.Spec.Role != models.HostRoleWorker || agent.Spec.ClusterDeploymentName == nil {
```

Comment:
you have it in 2 places, please move it to function.
@rollandf is this the right way to check if it is day2 cluster?